core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -194,10 +232,53 @@ class SYCLTeamMember {
    */
   template <typename Type>
   KOKKOS_INLINE_FUNCTION Type team_scan(const Type& value,
-                                        Type* const /*global_accum*/) const {
-    // FIXME_SYCL
-    Kokkos::abort("Not implemented!");
-    return value;
+                                        Type* const global_accum) const {
+    // We need to chunk up the whole reduction because we might not have
+    // allocated enough memory.
+    const int maximum_work_range =
+        std::min<int>(m_team_reduce_size / sizeof(Type), team_size());
+
+    int not_greater_power_of_two = 1;
+    while ((not_greater_power_of_two << 1) < maximum_work_range + 1)
+      not_greater_power_of_two <<= 1;
+
```

Comment:
Minor/non-blocking: did we ever decide on whether this should have curly braces? I prefer them, but if we haven't made it a policy then it's fine as is

Reply:
I don't think we have a policy. Otherwise, we should enforce it through `clang-format`.